.github/workflows/master.yml

```diff
@@ -60,23 +62,34 @@ jobs:
         uses: ./.github/actions/pip_installation
       - name: Serial C tests
         run: |
-          python -m pytest -rx -m "not parallel and c" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          python -m pytest -n auto -rx -m "not (parallel or xdist_incompatible) and c" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          pyccel-clean
+          python -m pytest -rx -m "xdist_incompatible and not parallel and c" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          pyccel-clean
         working-directory: ./tests
       - name: Serial fortran tests and non-tagged tests
         run: |
-          python -m pytest -rx -m "not parallel and not (c or python)" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          python -m pytest -n auto -rx -m "not (parallel or xdist_incompatible) and not (c or python)" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          pyccel-clean
+          python -m pytest -rx -m "xdist_incompatible and not parallel and not (c or python)" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          pyccel-clean
         working-directory: ./tests
       - name: Serial python tests
         run: |
-          python -m pytest -rx -m "not parallel and python" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          python -m pytest -n auto -rx -m "not (parallel or xdist_incompatible) and python" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          pyccel-clean
+          #python -m pytest -rx -m "xdist_incompatible and not parallel and python" --ignore=printers --ignore=symbolic --ignore=ndarrays
+          #pyccel-clean
         working-directory: ./tests
       - name: ND-Array tests
         run: |
-          python -m pytest ndarrays/ -rx
+          python -m pytest -n auto ndarrays/ -rx
+          pyccel-clean
         working-directory: ./tests
       - name: Parallel tests
         run: |
           mpiexec -n 4 python -m pytest epyccel/test_parallel_epyccel.py -v -m parallel -rx
+          pyccel-clean
```

Comment:
Same question as above

Reply:
When running in parallel with mpi4py, we have the same problems as with pytest-xdist so we don't clean up after parallel tests

Reply:
I think that there is a significant difference between running our "parallel-pyccel" tests and running the serial tests with `pytest-xdist`. In the first case only process #0 is actually compiling the files, while the other processes just load the shared library after it's been created. This means that we can easily add an MPI barrier to our tear-down functions and let process #0 take care of it. What do you think?

Reply:
Fixed in ab6cfe9